var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -276,6 +275,15 @@ def setup_build_environment(self, env):
         if self.spec.satisfies('os=bigsur'):
             env.set('SYSTEM_VERSION_COMPAT', 1)
 
+        # Force perl to build its own, internal copies of bzip2 and zlib.
+        # Otherwise, some modules can confuse config and break the build.
+        env.set('BUILD_BZIP2', 1)
+        env.unset('BZIP2_LIB')
+        env.unset('BZIP2_INCLUDE')
+        env.set('BUILD_ZLIB', 1)
+        env.unset('ZLIB_LIB')
+        env.unset('ZLIB_INCLUDE')
```

Comment:
Can we just unset the offending variables and add a dependency on `bzip2` and `zlib`?

Reply:
Almost.  You'd have to add `depends_on` and then set the environ vars to point to it.